### PR TITLE
feat: getFileStream() can accept non file request

### PR DESCRIPTION
**Pull request by fengmk2**
```js
const stream = await ctx.getFileStream({ required: false });
if (stream.filename) {
  console.log('uploaded ' + stream.filename);
} else {
  console.log('file not exists');
}
```

closes https://github.com/eggjs/egg/issues/2178

##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: app/extend/context.js [hunk 1]
```diff
@@ -28,15 +35,25 @@ module.exports = {
    * console.log(stream.fields);
    * ```
    * @method Context#getFileStream
+   * @param {Object} options
+   *  - {Boolean} options.required - required file submit, default is true
```
Comment (dead-horse) [inline app/extend/context.js]: requireFile ?

Comment (fengmk2) [reply app/extend/context.js]: 好

File: README.md [hunk 2]
```diff
@@ -150,6 +151,33 @@ module.exports = Class UploadController extends Controller {
       fields: stream.fields,
     };
   }
+
+  async uploadNotRequiredFile() {
+    const ctx = this.ctx;
+    // file not required
+    const stream = await ctx.getFileStream({ requireFile: false });
```
Comment (fengmk2) [inline README.md]: fixed

**Comment by fengmk2** [comment]
2.1.0

**Pull request merged**